odxtools/loadfile.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: MIT
+import os
+from pathlib import Path
+
+from .database import Database
+
+
+def load_pdx_file(pdx_file: str) -> Database:
+    db = Database()
+    db.add_pdx_file(pdx_file)
+    db.refresh()
+    return db
+
+
+def load_odx_d_file(odx_d_file_name: str) -> Database:
+    db = Database()
+    db.add_odx_file(odx_d_file_name)
+    db.refresh()
+
+    return db
+
+
+def load_file(file_name: str) -> Database:
+    if file_name.lower().endswith(".pdx"):
+        return load_pdx_file(file_name)
+    elif file_name.lower().endswith(".odx-d"):
+        return load_odx_d_file(file_name)
+    else:
+        raise RuntimeError(f"Could not guess the file format of file '{file_name}'!")
+
+
+def load_files(*file_names: str) -> Database:
+    db = Database()
+    for file_name in file_names:
+        p = Path(file_name)
+        if p.suffix.lower() == ".pdx":
+            db.add_pdx_file(file_name)
+        elif p.suffix.lower().startswith(".odx"):
+            db.add_odx_file(file_name)
+        elif p.name.lower() != "index.xml":
+            db.add_auxiliary_file(file_name)
+
+    db.refresh()
+    return db
+
+
+def load_directory(dir_name: str) -> Database:
```

Comment:
`load_files` can simply be extended to check if input file is a folder with `os.path.isdir`

When it comes to Linux philosophy, a directory is also a file

Reply:
I thought about this too, but I think it is better not to load directories recursively. (I guess that this would be a too easily accessible footgun...)

Reply:
I agree that `load_directory()` can be basically seen as a convenience wrapper around `load_files()`. using the same logic, `load_files()` can basically be seen as a convenience wrapper for `Database`, though?!